### ISSUE TITLE
Issue with SingleChildScrollView and BouncingScrollPhysics

### ISSUE THREAD

**Issue by armeaux** [ky1vstar/cupertino_interactive_keyboard#4]
There is an issue when wrapping it around a SingleChildScrollView with BouncingScrollPhysics. When the scrollposition is near the top, and the keyboard is being dismissed interactively, it keeps jumping back to offset 0 for some reason.

**Comment by ky1vstar** [comment]
@armeaux thanks for reporting this issue. Could you please provide the video or reproducible example?

**Comment by armeaux** [comment]
@ky1vstar I added a video below, however, the glitch flashes too quickly on a physical device to be properly by captured by a screenrecording.

https://github.com/ky1vstar/cupertino_interactive_keyboard/assets/24753532/a3a4939e-0b8a-4b0e-9480-e6d0d0570f81

Here is the code (should be reproducible)

```
import 'package:cupertino_interactive_keyboard/cupertino_interactive_keyboard.dart';
import 'package:flutter/material.dart';

class KeyboardTestView extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text("Keyboard Bug")),
      body: Container(  
        color: Colors.blue,
        height: MediaQuery.of(context).size.height,
        width: MediaQuery.of(context).size.width,
        child: CupertinoInteractiveKeyboard(
          child: SingleChildScrollView(  
            physics: BouncingScrollPhysics(),
            child: Container( 
              color: Colors.white, 
              padding: EdgeInsets.only(top: 100),
              height: MediaQuery.of(context).size.height + 10,              
              child: TextField(maxLines: 10),
            )
          ),
        ),
      ),
    );
  }
}
```

Since ListViews tend to rebuild widgets, my controllers get disposed prematurely. SingleChildScrollView does not appear to have this, so I need to use one of those. Right now i am changing the ScrollPhysics from Bouncing to Clamped as soon as the TextField gets a focus. This way it doesn't glitch, however, it also does not provide the native iOS experience. 
I have tried figuring out what causes this glitch but to no avail. 

If there is something i can do to help, feel free to let me know! 

EDIT: sorry, didn't check the screenrecording well enough, the glitch is clearly visible!

**Comment by ky1vstar** [comment]
@armeaux I guess you are using iOS 17, because I cannot reproduce it on iOS 16.6. I'll try it a little bit later this week

**Comment by armeaux** [comment]
@ky1vstar With this phone i am actually on iOS 16.1.2
I have other phones with other iOS versions I could try it on, but it seems like an issue related to Flutter's behavior. 

It appears that the ScrollController position is set to 0 during every rebuild of the UI. I have been lurking around in the code for CustomScrollView to find possible reasons. Other Scrollable (ListView, CustomScrollView) widgets do not seem to suffer from this issue.

Will try to give it some time, and if i figure something out myself i will keep you updated in this thread. 

EDIT 1: During the animation, the ScrollController seems to detach/attach during every build. 
This does not happen for the ListView.

**Comment by orestesgaolin** [comment]
Similar glitching appears on iOS 18 with the example app

https://github.com/user-attachments/assets/604bc4d9-fa35-44ed-b81c-fa901747e831

**Comment by ky1vstar** [comment]
> Similar glitching appears on iOS 18 with the example app
> 
>  ScreenRecording_11-22-2024.10-49-38.PM_11.mp4

@orestesgaolin initial issue in this tread is related to unique behaviour of SingleChildScrollView.

Issue in the video is caused by the delay between updated keyboard position is reported to Flutter framework on native part and it is being actually rendered. This delay relates to asynchronous manner of Flutter in general, and can only be fixed by the Flutter team(

**Comment by orestesgaolin** [comment]
I see, so perhaps threads merging and synchronous access to native apis might help here?

**Comment by ky1vstar** [comment]
@orestesgaolin yes, it definitely would. Not sure, if it is already supported by the framework.

**Comment by orestesgaolin** [comment]
From what I've heard at fluttercon, it's one of the big goals for the next year, especially in the context of platform views
